### PR TITLE
Add templates

### PR DESCRIPTION
**Pull request by MarkSalabutin**

### CODE REVIEW

File: cmd/mgo/main.go [hunk 1]
```diff
@@ -28,7 +29,21 @@ func initDB() model.Router {
 		appStorage,
 		userStorage,
 	)
-	r := ihttp.NewRouter(nil, appStorage, userStorage, tokenStorage, tokenService)
+
+	staticPages := model.StaticPages{
```
Comment (erudenko) [inline cmd/mgo/main.go]: static pages don't have any relation to database, CORS as well.

File: http/api_router.go [hunk 2]
```diff
@@ -30,13 +32,21 @@ func (ar *apiRouter) ServeHTTP(w http.ResponseWriter, r *http.Request) {
...
 //NewRouter created and initiates new router
-func NewRouter(logger *log.Logger, appStorage model.AppStorage, userStorage model.UserStorage, tokenStorage model.TokenStorage, tokenService model.TokenService) model.Router {
+func NewRouter(logger *log.Logger, appStorage model.AppStorage, userStorage model.UserStorage, tokenStorage model.TokenStorage, tokenService model.TokenService, options ...func(*apiRouter) error) (model.Router, error) {
 	ar := apiRouter{
 		router:       negroni.Classic(),
 		appStorage:   appStorage,
 		userStorage:  userStorage,
 		tokenStorage: tokenStorage,
 		tokenService: tokenService,
+		handler:      mux.NewRouter().StrictSlash(true),
```
Comment (erudenko) [inline http/api_router.go]: please don't use strict slash

File: cmd/mgo/main.go [hunk 3]
```diff
@@ -40,7 +40,7 @@ func initServices() (model.AppStorage, model.UserStorage, model.TokenStorage, mo
 func initRouter() model.Router {
 	appStorage, userStorage, tokenStorage, tokenService := initServices()
 
-	router, err := ihttp.NewRouter(nil, appStorage, userStorage, tokenStorage, tokenService)
+	router, err := ihttp.NewRouter(nil, appStorage, userStorage, tokenStorage, tokenService, ihttp.ServeDefaultStaticPages())
```
Comment (erudenko) [inline cmd/mgo/main.go]: this pages should be served without any additional settings, because they are required

File: http/serve_static.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+func ServeStaticPages(sp StaticPages) func(*apiRouter) error {
+	return func(ar *apiRouter) error {
+		return ar.serveStaticPages(sp)
+	}
+}
+
+// ServeDefaultStaticPages serves default HTML pages
+func ServeDefaultStaticPages() func(*apiRouter) error {
+	staticPages := StaticPages{
+		Login:          "../../static/login.html",
```
Comment (erudenko) [inline http/serve_static.go]: I don't think it's a good default path. I recommend using subfolder in the working directory, like: `./static/login.html`

File: http/serve_static.go [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+		Registration:   "../../static/registration.html",
+		ForgotPassword: "../../static/forgot-password.html",
+		ResetPassword:  "../../static/reset-password.html",
+	}
+
+	return ServeStaticPages(staticPages)
+}
+
+func (ar *apiRouter) serveStaticPages(sp StaticPages) error {
+	ar.handler.HandleFunc("/login", ar.ServeTemplate(sp.Login)).Methods("GET")
```
Comment (erudenko) [inline http/serve_static.go]: please use regexp to handle trailing slash

File: http/serve_static.go (hunk 4 above)
Comment (MarkSalabutin) [reply http/serve_static.go]: We have 4 different working directories for each persistent layer, so I have to copy folder `static` into each working directory to make it work correctly.

**Pull request merged**